imblearn/under_sampling/_prototype_selection/_instance_hardness_threshold.py

```diff
@@ -116,11 +116,7 @@ def __init__(
     def _validate_estimator(self, random_state):
         """Private function to create the classifier"""
 
-        if (
-            self.estimator is not None
-            and isinstance(self.estimator, ClassifierMixin)
-            and hasattr(self.estimator, "predict_proba")
-        ):
+        if self.estimator is not None and hasattr(self.estimator, "predict_proba"):
```

Comment:
This change removes the `isinstance` check for `InstanceHardnessThreshold`, similar to how [`KMeansSMOTE`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/f177b05256fa3e4e556c1a8be0283b721b1bcd86/imblearn/over_sampling/_smote/cluster.py#L148-L158) does not require such a check.